STM32/Temperature/Core/Src/Temperature.c

```diff
@@ -81,18 +100,40 @@ static int initSpiDevices(SPI_HandleTypeDef* hspi)
     {
         int ret = ADS1120Init(&ads1120[i]);
         if (ret != 0) {
+
             err |= ret << (4*i);
 
             // If connection could not be established to chip
             // set temperatures to 10010 to indicate miscommunication
         	ads1120[i].data.internalTemp = 10010;
             ads1120[i].data.chA = 10010;
             ads1120[i].data.chB = 10010;
+
+            // The TEMP_ADS1120_Error_Msk maps to the bit relating to the
+            // 0th ADS1120 on the temperature board. Hence, we shift the index
+            // according to the relevant chip having an error.
+            bsSetError( ret << (i*2) );
```

Comment:
At the moment, any error will persist until the board is reset. I guess since this is the initialisation and there is no possibility of re-initialisation, this is probably OK, but worth thinking about in the future.

Is it possible for any errors to happen in the loop?

Reply:
You are right! Will update the PR with a actual on-going monitoring of the system states shortly.